uplink/storage/streams/store.go

```diff
@@ -418,42 +413,38 @@ func (s *streamStore) Delete(ctx context.Context, path Path, pathCipher storj.Ci
 		return err
 	}
 
-	lastSegmentPath, err := createSegmentPath(ctx, -1, path.Bucket(), encPath)
-	if err != nil {
-		return err
-	}
+	// TODO maybe for now just use GetObject and decrypt num of segments from metainfo
+	// to avoid two satellite requestes ??
```

Comment:
The reason we don't want to ask a "number of segments" is because with multipart uploads we won't know the part numbers, so we would end up needing to do the listing anyways. But we could batch begin delete and list segments together into a single request.

Reply:
yeah, this was only for the current implementation. For batching we still need to implement passing params between requests/response but yes this can be done.